mls-rs-codec/src/cow.rs

```diff
@@ -0,0 +1,31 @@
+use alloc::borrow::Cow;
+
+use crate::{Error, MlsDecode, MlsEncode, MlsSize};
+
+impl<'a, T> MlsSize for Cow<'a, T>
+where
+    T: MlsSize + ToOwned,
+{
+    fn mls_encoded_len(&self) -> usize {
+        self.as_ref().mls_encoded_len()
+    }
+}
+
+impl<'a, T> MlsEncode for Cow<'a, T>
+where
+    T: MlsEncode + ToOwned,
+{
+    #[inline]
+    fn mls_encode(&self, writer: &mut Vec<u8>) -> Result<(), Error> {
+        self.as_ref().mls_encode(writer)
+    }
+}
+
+impl<'a, T> MlsDecode for Cow<'a, T>
+where
+    T: MlsDecode + ToOwned<Owned = T>,
```

Comment:
I think the proper bound should be:
```suggestion
    T: ToOwned,
    <T as ToOwned>::Owned: MlsDecode,
```

This way this implementation also applies to types like `Cow<'_, str>` where the owned and borrowed types are different. We only care about the owned type being decodable here.